onnx/backend/test/runner/__init__.py

```diff
@@ -215,13 +215,15 @@ def assert_similar_outputs(
                     if ref_outputs[i].tolist() != outputs[i].tolist():
                         raise AssertionError(f"{ref_outputs[i]} != {outputs[i]}")
                     continue
+                np.testing.assert_equal(outputs[i].dtype, ref_outputs[i].dtype)
+                np.testing.assert_array_equal(outputs[i].shape, ref_outputs[i].shape)
```

Comment:
Probably good to provide a message telling users this is testing for the shape

Reply:
Thanks for the suggestion, just pushed a fix. Here is how it looks like in a case of failure after the last commit:
```console
E   AssertionError:
E   Arrays are not equal
E   Output 0 has incorrect shape
E   (shapes (0,), (3,) mismatch)
E    x: array([], dtype=float64)
E    y: array([3, 4, 5])
```